cmd/editorconfig-checker/main_test.go

```diff
@@ -0,0 +1,161 @@
+	mainHasRun <- code
+	runtime.Goexit()
+}
+
+func setArguments(t *testing.T, args ...string) {
+	t.Helper()
+
+	initialArgs := os.Args
+	t.Cleanup(func() {
+		os.Args = initialArgs
+	})
+
+	// os.Args needs to have the binary as the first element
+	os.Args = append([]string{"editorconfig-checker"}, args...)
+}
+
+func runWithArguments(t *testing.T, args ...string) (string, int) {
+	t.Helper()
+	setArguments(t, args...)
+	outputBuffer := new(bytes.Buffer)
+	loggerInjectionHook = func() {
+		currentConfig.Logger.SetWriter(outputBuffer)
+	}
+	go main()
+	exitCode := <-mainHasRun // must not be inlined into the return statement, since we need to wait for main() to exist before trying to read the buffer
+	return outputBuffer.String(), exitCode
+}
+
+func TestMainOurCodebase(t *testing.T) {
+	cdRelativeToRepo(t, "")
+	_, lastSeenCode := runWithArguments(t, "--debug", "--verbose", "--exclude", `\.git`, "--exclude", `\.exe$`)
+	if lastSeenCode != exitCodeNormal {
+		t.Errorf("main exited with return code %d, but we expected %d", lastSeenCode, exitCodeNormal)
+	}
+}
+
+func TestMainMissingExplicitConfig(t *testing.T) {
+	cdRelativeToRepo(t, "")
+	_, lastSeenCode := runWithArguments(t, "--debug", "--verbose", "--exclude", `\.git`, "--exclude", `\.exe$`, "--config", "/nonexistant")
+	if lastSeenCode != exitCodeConfigFileNotFound {
+		t.Errorf("main exited with return code %d, but we expected %d", lastSeenCode, exitCodeConfigFileNotFound)
+	}
+}
+func TestMainWithFilesGiven(t *testing.T) {
+	cdRelativeToRepo(t, "")
+	_, lastSeenCode := runWithArguments(t, "--debug", "--verbose", "README.md")
+	if lastSeenCode != exitCodeNormal {
+		t.Errorf("main exited with return code %d, but we expected %d", lastSeenCode, exitCodeNormal)
+	}
+}
+
+func TestMainInitializingANewConfig(t *testing.T) {
+	dir := t.TempDir()
+
+	_, lastSeenCode := runWithArguments(t, "--debug", "--verbose", "--init", "--config", dir+"/testwriteconfig.json")
+	if lastSeenCode != exitCodeNormal {
+		t.Errorf("main exited with return code %d, but we expected %d", lastSeenCode, exitCodeNormal)
+	}
+
+	// do the same test a second time, trying to "overwrite" the existing file
+	_, lastSeenCode = runWithArguments(t, "--debug", "--verbose", "--init", "--config", dir+"/testwriteconfig.json")
+	// but now we expect it to fail since it does not want to overwrite the existing file
+	if lastSeenCode != exitCodeErrorOccurred {
+		t.Errorf("main exited with return code %d, but we expected %d", lastSeenCode, exitCodeErrorOccurred)
+	}
+}
+
+func TestMainLoadingAncientConfig(t *testing.T) {
+	output, lastSeenCode := runWithArguments(t, "--debug", "--verbose", "--config", "testdata/ancient-config.json")
+	if lastSeenCode != exitCodeErrorOccurred {
+		t.Errorf("main exited with return code %d, but we expected %d", lastSeenCode, exitCodeErrorOccurred)
+	}
+	if !strings.Contains(output, "SpacesAftertabs") {
+		t.Errorf("main did not produce a warning that SpacesAftertabs is deprecated\nOutput:\n%s", output)
```

Comment:
I'm surprised it's SpaceAftertabs and not SpaceAfterTabs

But I assume it's out of the PR scope

Reply:
we're testing that  the deprecation warning is output, not the presence of the new option.